src/main/java/de/tum/in/www1/artemis/service/EmailSummaryService.java

```diff
@@ -0,0 +1,156 @@
+     * @param user for whom the summary email should be prepared
+     */
+    private void prepareEmailSummaryForUser(User user) {
+        // Get all courses with exercises, lectures and exams (filtered for given user)
+        List<Course> courses = courseService.findAllActiveWithExercisesAndLecturesAndExamsForUser(user);
+
+        // Get released exercises of the last week (which are not yet over, i.e. working due date is not passed yet)
+        Set<Exercise> newExercises = getAllExercisesForSummary(courses);
+
+        // More elements that should be displayed in weekly summaries can be extracted here
+        // Currently only exercises are used for weekly summaries
+
+        // currently, only weekly summaries are supported -> for daily just add one more case
+        if (scheduleInterval.equals(weekly)) {
+            mailService.sendWeeklySummaryEmail(user, newExercises);
+        }
+    }
+
+    /**
+     * Filter users based on their user/notification settings and schedule interval
+     * e.g. if the schedule interval is weekly -> check for weekly summary setting
+     * @param user who should be checked
+     * @return true if the setting that corresponds to the used interval is activated else false
+     */
+    private boolean filterBasedOnUserSettingAndScheduleInterval(User user) {
+        // currently, only weekly summaries are supported -> for daily just add one more case
+        if (scheduleInterval.equals(weekly)) {
+            return checkIfWeeklySummaryIsAllowedByNotificationSettingsForGivenUser(user);
+        }
+        else
+            throw new UnsupportedOperationException("Unsupported scheduleInterval: " + scheduleInterval);
+    }
+
+    /**
+     * @param courses which exercises will be extracted
+     * @return all still active exercises based on the students courses and the scheduleInterval
+     * e.g. scheduleInterval = 7 days -> all still active exercises of the last week (7 days)
+     */
+    private Set<Exercise> getAllExercisesForSummary(List<Course> courses) {
+        Set<Exercise> newExercises = new HashSet<>();
+        courses.forEach(course -> newExercises.addAll(course.getExercises().stream().filter(this::shouldExerciseBePartOfEmailSummary).collect(Collectors.toSet())));
+        return newExercises;
+    }
+
+    /**
+     * Checks if an exercise should be part of the email summary
+     * Exercise should have been released, not yet ended, and the release should have been in the time frame of now - scheduleInterval
+     *
+     * @param exercise that should be checked
+     * @return true if the exercise should be part of the email summary else false
+     */
+    private boolean shouldExerciseBePartOfEmailSummary(Exercise exercise) {
+        if (exercise == null || exercise.getReleaseDate() == null) {
+            return false;
+        }
+        return exercise.isReleased() && exercise.getReleaseDate().isAfter(ZonedDateTime.now().minus(scheduleInterval)) && !exercise.isEnded();
```

Comment:
This will break when merging, as the `exercise.inEnded()` method no longer exists.

I think just a simple check if `ZonedDateTime.now().isBefore(exercise.getDueDate())` should be enough. Alternatively, if you care about the individual due dates, you can use `ExerciseDateService::isBeforeLatestDueDate`.

Reply:
Thank you very much for pointing this out. I replaced the method call with your first recommendation. 👍